GameOfLifeApp/GameLogic.cs

```diff
@@ -0,0 +1,160 @@
+            CurrentField[2, 0] = true;
+            CurrentField[1, 1] = true;
+            CurrentField[2, 1] = true;
+            CurrentField[1, 2] = true;
+        }
+
+        /// <summary>
+        /// Gets number of alive cells around a given cell. 
+        /// </summary>
+        /// <param name="currentRow">Row of a current cell.</param>
+        /// <param name="currentColumn">Column of a current cell.</param>
+        /// <returns>Return number of alive neighbours.</returns>
+        private int NeighboursCount(int currentRow, int currentColumn)
+        {
+            int count = 0;
+            for (int row = currentRow - 1; row <= currentRow + 1; row++)
+            {
+                for (int column = currentColumn - 1; column <= currentColumn +1; column++)
+                {
+                    int actualRow;
+                    int actualColumn;
+                    if (row < 0)
+                    {
+                        actualRow = CurrentField.GetLength(0) - 1;
+                    }
+                    else if (row == CurrentField.GetLength(0))
+                    {
+                        actualRow = 0;
+                    }
+                    else
+                    {
+                        actualRow = row;
+                    }
+                    if (column < 0)
+                    {
+                        actualColumn = CurrentField.GetLength(1) - 1;
+                    }
+                    else if (column == CurrentField.GetLength(1))
+                    {
+                        actualColumn = 0;
+                    }
+                    else
+                    {
+                        actualColumn = column;
+                    }
+                    count += CurrentField[actualRow, actualColumn] ? 1 : 0;
+                }
+            }
+            count -= CurrentField[currentRow, currentColumn] ? 1 : 0;
+
+            return count;
+        }
+
+        /// <summary>
+        /// Gets new boolean values according to statements.
+        /// </summary>
+        /// <param name="neighboursCount">Number of alive cells around a given cell.</param>
+        /// <param name="isCurrentAlive">Given cell is alive. Value of isCurrentAlive: true.</param>
+        /// <returns>Return the next value for a given cell.</returns>
+        private static bool GetCellOffsprings(int neighboursCount, bool isCurrentAlive)
+        {
+            if (!isCurrentAlive && neighboursCount == 3)
+            {
+                return true;
+            }
+            else if (isCurrentAlive && neighboursCount == 2 || neighboursCount == 3)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// This method generates a new field and transfers data to a generated field.       
+        /// </summary>
+        public void GetNextGeneration()
+        {
+            bool[,] generatedField2 = new bool[CountOfRows, CountOfColumns];
```

Comment:
Now that you use this local variable in the method, you no longer need the class property `readonly bool[,] generatedField;` and it can be removed. When you remove it, you can then rename this `generatedField2` to `generatedField`

Reply:
I use public bool[,] GeneratedField for  the initial field.

Reply:
Yes, you generate your initial field in `GeneratedField ` property, but you also instantly assign the `GeneratedField` value to `CurrentField`, meaning, that your `CurrentField` property is also holding the same value (The initial field value) and you can use that instead.